mlx/fast_primitives.h

```diff
@@ -65,4 +65,39 @@ class RoPE : public Custom {
   int offset_;
 };
 
+class ScaledDotProductAttention : public Custom {
+ public:
+  explicit ScaledDotProductAttention(
+      Stream stream,
+      std::function<std::vector<array>(std::vector<array>)> fallback,
+      const float scale,
+      const bool needs_mask)
+      : Custom(stream, fallback), scale_(scale), needs_mask_(needs_mask){};
+
+  void eval_cpu(const std::vector<array>& inputs, std::vector<array>& outputs)
+      override {
+    return eval_cpu(inputs, outputs[0]);
```

Comment:
```suggestion
    outputs[0] = fallback_(inputs)[0];
```

(and delete the extra `eval_cpu` below